src/AssetRegistry/ATokenSubRegistry.sol

```diff
@@ -0,0 +1,185 @@
+ */
+pragma solidity >=0.4.22 <0.9.0;
+
+import "./AbstractSubRegistry.sol";
+import "../interfaces/IAToken.sol";
+import {FixedPointMathLib} from "../utils/FixedPointMathLib.sol";
+
+/**
+ * @title Sub-registry for Standard ERC20 tokens
+ * @author Arcadia Finance
+ * @notice The StandardERC20Registry stores pricing logic and basic information for ERC20 tokens for which a direct price feed exists
+ * @dev No end-user should directly interact with the StandardERC20Registry, only the Main-registry, Oracle-Hub or the contract owner
+ */
+contract ATokenSubRegistry is SubRegistry {
+    using FixedPointMathLib for uint256;
+
+    struct AssetInformation {
+        uint64 assetUnit;
+        address assetAddress;
+        address underlyingAssetAddress;
+        address[] underlyingAssetOracleAddresses;
+    }
+
+    mapping(address => AssetInformation) public assetToInformation;
+
+    /**
+     * @notice A Sub-Registry must always be initialised with the address of the Main-Registry and of the Oracle-Hub
+     * @param mainRegistry The address of the Main-registry
+     * @param oracleHub The address of the Oracle-Hub
+     */
+    constructor(address mainRegistry, address oracleHub)
+        SubRegistry(mainRegistry, oracleHub)
+    {}
+
+    /**
+     * @notice Adds a new asset to the StandardERC20Registry, or overwrites an existing asset.
+     * @param assetInformation A Struct with information about the asset
+     *                         - assetUnit: The unit of the asset, equal to 10 to the power of the number of decimals of the asset
+     *                         - assetAddress: The contract address of the asset
+     *                         - oracleAddresses: An array of addresses of oracle contracts, to price the asset in USD
+     * @param assetCreditRatings The List of Credit Ratings for the asset for the different BaseCurrencies.
+     * @dev The list of Credit Ratings should or be as long as the number of baseCurrencies added to the Main Registry,
+     *      or the list must have length 0. If the list has length zero, the credit ratings of the asset for all baseCurrencies is
+     *      is initiated as credit rating with index 0 by default (worst credit rating).
+     * @dev The assets are added/overwritten in the Main-Registry as well.
+     *      By overwriting existing assets, the contract owner can temper with the value of assets already used as collateral
+     *      (for instance by changing the oracleaddres to a fake price feed) and poses a security risk towards protocol users.
+     *      This risk can be mitigated by setting the boolean "assetsUpdatable" in the MainRegistry to false, after which
+     *      assets are no longer updatable.
+     * @dev Assets can't have more than 18 decimals.
+     */
+    function setAssetInformation(
+        AssetInformation calldata assetInformation,
+        uint256[] calldata assetCreditRatings
+    ) external onlyOwner {
+        IOraclesHub(oracleHub).checkOracleSequence(
```

Comment:
i think it might be possible to fetch all information directly from the StandardERC20Subregistry, since we will not allow an aToken for which we don't have the underlying asset as well as collateral, decreases probability on errors

Reply:
done